announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java

```diff
@@ -408,12 +409,57 @@ private String getAddress(Event event, AddressField field)
 	}
 
 	/**
-	 * Add to the user list any other users who should be notified about this ref's change.
+	 * @inheritDoc
+	 */
+	protected List<User> getRecipients(Event event)
+	{
+		Reference ref = entityManager.newReference(event.getResource());
+		String siteId = (getSite() != null) ? getSite() : ref.getContext();
+		try {
+			Site site = siteService.getSite(siteId);
+			List<User> users = super.getRecipients(event);
+
+			// remove any other users
+			removeSpecialRecipients(users, ref, site);
+			return users;
+		} catch (Exception any) {
+			return new Vector<User>();
+		}
+	}
+
+	/**
+	 * Remove in the user list any other users who shouldn't be notified
 	 * 
 	 * @param users
-	 *        The user list, already populated based on site visit and resource ability.
+	 * 		  a list of users for message
 	 * @param ref
 	 *        The entity reference.
+	 * @param site
+	 * 		The site whose membership the users will be deleted to
+	 */
+	protected void removeSpecialRecipients(List<User> users, Reference ref, Site site)
+	{
+		final AnnouncementMessage msg = (AnnouncementMessage) ref.getEntity();
+
+		for (Iterator<User> i = users.listIterator(); i.hasNext();) {
```

Comment:
Let's use filters and Collectors to cleanup the list!

Reply:
Or maybe do it with a for loop setting a new list of allowed users, makes sense?